script/tool/lib/src/pathify_command.dart

```diff
@@ -0,0 +1,244 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:flutter_plugin_tools/src/common/repository_package.dart';
+import 'package:git/git.dart';
+import 'package:path/path.dart' as p;
+import 'package:pub_semver/pub_semver.dart';
+import 'package:pubspec_parse/pubspec_parse.dart';
+
+import 'common/core.dart';
+import 'common/git_version_finder.dart';
+import 'common/plugin_command.dart';
+
+const int _exitPackageNotFound = 3;
+const int _exitCannotUpdatePubspec = 4;
+
+/// Converts all dependencies on target packages to path-based dependencies.
+///
+/// This is to allow for pre-publish testing of changes that could affect other
+/// packages in the repository. For instance, this allows for catching cases
+/// where a non-breaking change to a platform interface package of a federated
+/// plugin would cause post-publish analyzer failures in another package of that
+/// plugin.
+class PathifyCommand extends PluginCommand {
+  /// Creates an instance of the pathify command.
+  PathifyCommand(
+    Directory packagesDir, {
+    GitDir? gitDir,
+  }) : super(packagesDir, gitDir: gitDir) {
+    argParser.addMultiOption(_targetPackagesArg,
+        help: 'The names of the packages to convert to dependencies.\n'
```

Comment:
This is recursive, right?  We should mention that.

Reply:
It's not recursive; only direct dependencies are handled. There's no need for it to be recursive because the whole repository is updated; any transitive dependencies will be through other packages in the repository.